server/mailrepository/mailrepository-memory/src/main/java/org/apache/james/mailrepository/memory/MemoryMailRepository.java

```diff
@@ -26,16 +26,18 @@
 
 import javax.mail.MessagingException;
 
+import org.apache.james.lifecycle.api.LifecycleUtil;
 import org.apache.james.mailrepository.api.MailKey;
 import org.apache.james.mailrepository.api.MailRepository;
 import org.apache.mailet.Mail;
 
 public class MemoryMailRepository implements MailRepository {
 
-    private final ConcurrentHashMap<MailKey, Mail> mails;
+    private static final ConcurrentHashMap<MailKey, Mail> mails = new ConcurrentHashMap<>();
```

Comment:
No all mail repository should not share a common memory stucture

And this static field cheats by preventing GC to find those.

What we should likely do is:

 - Always dispose emails upon delete, clear
 - Call clear in our @AfterEach callbacks